app/models/commodity.rb

```diff
@@ -187,6 +187,14 @@ def goods_nomenclature_class
     declarable? ? 'Commodity' : 'Subheading'
   end
 
+  def special_nature?
+    measures.any? { |m| m.measure_conditions.map(&:certificate_type_code).include?(Certificate::SPECIAL_NATURE) }
```

Comment:
I think it makes sense to add these to the commodity presenter which cached the result of calling the massively inefficient measures dataset. The measures dataset isn't actually eager loadable _and_ it does a whole bunch of unions to work out the start and end dates of a measure.

https://github.com/trade-tariff/trade-tariff-backend/blob/main/app/presenters/api/v2/commodities/commodity_presenter.rb#L14-L23

Reply:
Another thing to know is that preference codes only apply to import measures as they're part of your declarations